analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/UsingNonstandardCryptographyBase.cs

```diff
@@ -30,10 +30,10 @@ public abstract class UsingNonstandardCryptographyBase<TSyntaxKind, TTypeDeclara
         where TTypeDeclarationSyntax : SyntaxNode
         where TSyntaxKind : struct
```

Comment:
Can we make this an enum constraint?

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters#enum-constraints

Reply:
No, it boils down to Roslyn API that has a `struct` constraint in `Microsoft.CodeAnalysis.Diagnostics.AnalysisContext.RegisterSyntaxNodeAction`
```
        public void RegisterSyntaxNodeAction<TLanguageKindEnum>(Action<SyntaxNodeAnalysisContext> action, params TLanguageKindEnum[] syntaxKinds) where TLanguageKindEnum : struct

```